state/protocol/badger/snapshot.go

```diff
@@ -221,10 +221,10 @@ func (s *Snapshot) SealingSegment() (*flow.SealingSegment, error) {
 	}
 
 	// STEP (iii): extended history to allow checking for duplicated collections, i.e.
-	// limitHeight = max(head.Height - flow.DefaultTransactionExpiry, SporkRootBlockHeight)
+	// limitHeight = max(blockSealedAtHead.Height - flow.DefaultTransactionExpiry, SporkRootBlockHeight)
 	limitHeight := s.state.sporkRootBlockHeight
-	if head.Header.Height > s.state.sporkRootBlockHeight+flow.DefaultTransactionExpiry {
-		limitHeight = head.Header.Height - flow.DefaultTransactionExpiry
+	if blockSealedAtHead.Height > s.state.sporkRootBlockHeight+flow.DefaultTransactionExpiry {
```

Comment:
Let me explain why changing this:

When using `head.Header.Height`, the number of blocks gets included in extra blocks is less than `600` (DefaultTransactionExpiry), why? 
For instance, let's say block 800 seals block 792, if the create protocol snapshot for block 800, then the SealingSegment.Blocks will include 8 blocks `[block_792, block_800]`, and the ExtraBlocks field will include 592 blocks `[block_200, block_791]`, since `block_792` is the last sealed block, there are 592 blocks below the sealed block, which is not enough (we need 600). why?

Because, if an execution node is dynamically bootstrapped, it will re-execute from next unsealed block. Let's use the above example again, if we bootstrap an execution node with the protocol snapshot for block 800, then it will re-execute blocks `[block_792, block_800]`, if a transaction in `block_792` tries to query block at height 192 (792 - 600 = 192), then it will fail, because the protocol snapshot only includes block starting from height 200. In order to prevent this error, we would have to take the `8` blocks into consideration, which is a bit complex. A simpler approach is just to index a bit more blocks by indexing 600 blocks below the blockSealedAtHead, which means we will include exactly 600 blocks `[block_192, block791]` in the ExtraBlocks field to be indexed. And then, it will guarantee the block height limit we set on getBlock query will always work.